src/input_stream.rs

```diff
@@ -0,0 +1,270 @@
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        let buffer_ptr = buffer.as_mut_ptr();
+        let capacity = buffer.capacity();
+        unsafe {
+            let mut bytes_read = mem::uninitialized();
+            let mut error = ptr::null_mut();
+            let _ = ffi::g_input_stream_read_all(self.to_glib_none().0, buffer_ptr, capacity, &mut bytes_read, cancellable.0, &mut error);
+
+            assert!(buffer.capacity() >= bytes_read);
+            buffer.set_len(bytes_read);
+
+            if error.is_null() {
+                Ok((buffer, None))
+            } else if bytes_read != 0 {
+                Ok((buffer, Some(from_glib_full(error))))
+            } else {
+                Err((buffer, from_glib_full(error)))
+            }
+        }
+    }
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn read_all_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<(Vec<u8>, Option<Error>), (Vec<u8>, Error)>) + Send + 'static>(&self, mut buffer: Vec<u8>, io_priority: Priority, cancellable: P, callback: Q) {
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        let buffer_ptr = buffer.as_mut_ptr();
+        let capacity = buffer.capacity();
+        let user_data: Box<Option<(Box<Q>, Vec<u8>)>> = Box::new(Some((Box::new(callback), buffer)));
+        unsafe extern "C" fn read_all_async_trampoline<Q: FnOnce(Result<(Vec<u8>, Option<Error>), (Vec<u8>, Error)>) + Send + 'static>(_source_object: *mut gobject_ffi::GObject, res: *mut ffi::GAsyncResult, user_data: glib_ffi::gpointer)
+        {
+            callback_guard!();
+
+            let mut user_data: Box<Option<(Box<Q>, Vec<u8>)>> = Box::from_raw(user_data as *mut _);
+            let (callback, mut vec) = user_data.take().unwrap();
+
+            let mut error = ptr::null_mut();
+            let mut bytes_read = mem::uninitialized();
+            let _ = ffi::g_input_stream_read_all_finish(_source_object as *mut _, res, &mut bytes_read, &mut error);
+
+            assert!(vec.capacity() >= bytes_read);
+            vec.set_len(bytes_read);
+
+            let result = if error.is_null() {
+                Ok((vec, None))
+            } else if bytes_read != 0 {
+                Ok((vec, Some(from_glib_full(error))))
+            } else {
+                Err((vec, from_glib_full(error)))
+            };
+
+            callback(result);
+        }
+        let callback = read_all_async_trampoline::<Q>;
+        unsafe {
+            ffi::g_input_stream_read_all_async(self.to_glib_none().0, buffer_ptr, capacity, io_priority.to_glib(), cancellable.0, Some(callback), Box::into_raw(user_data) as *mut _);
+        }
+    }
+
+    fn read_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<Vec<u8>, (Vec<u8>, Error)>) + Send + 'static>(&self, mut buffer: Vec<u8>, io_priority: Priority, cancellable: P, callback: Q) {
```

Comment:
@sdroege Can you check this function, tests with it failing with AV.
Seems I made some mistake, but can't find it.